src/python/WMCore/Services/WMArchive/DataMap.py

```diff
@@ -52,7 +54,8 @@
                                 'readTotalMB': float,
                                 'readTotalSecs': float,
                                 'writeTotalMB': float,
-                                'writeTotalSecs': float}}
+                                'writeTotalSecs': float},
+                    'cmssw': CMSSWMetrics()}
```

Comment:
Do I understand it right that this function would be executed only once during the lifetime of the ArchiveDataReporter? Which is the component responsible for parsing the FJR and creating the WMArchive document.

Reply:
I think it will be done once, but is hard to say based on my current understanding of the all involved components. The parsing will be done by `WMCore/FwkJobReport/XMLParser.py` when it will parse provided CMSSW XML FJR file, i.e. it will be done in `WMCore/WMSpec/Steps/Executors/CMSSW.py` module and whoever component will call it. But it is not happen in ArchiveDataReporter.

Reply:
I was afraid it could happen for every single job, but grepping the code:
```
(py3.8) OR-FVFH37J7Q6LR:WMCore amaltar2$ grep -rI DataMap src/* | grep import
src/python/WMComponent/ArchiveDataReporter/ArchiveDataPoller.py:from WMCore.Services.WMArchive.DataMap import createArchiverDoc
```

my understand is that the map will be created once, the moment ArchiveDataReporter component gets started.
In that case, everything is fine.